pkg/cloudprovider/backup_service.go

```diff
@@ -137,12 +143,12 @@ func (br *backupService) UploadBackup(bucket, backupName string, metadata, backu
 	return nil
 }
 
-func (br *backupService) DownloadBackup(bucket, backupName string) (io.ReadCloser, error) {
-	return br.objectStorage.GetObject(bucket, getBackupContentsKey(backupName))
+func (br *backupService) DownloadBackup(bucket, bucketPath, backupName string) (io.ReadCloser, error) {
+	return br.objectStorage.GetObject(bucket, getBackupContentsKey(bucketPath, backupName))
 }
 
-func (br *backupService) GetAllBackups(bucket string) ([]*api.Backup, error) {
-	prefixes, err := br.objectStorage.ListCommonPrefixes(bucket, "/")
+func (br *backupService) GetAllBackups(bucket string, bucketPath string) ([]*api.Backup, error) {
+	prefixes, err := br.objectStorage.ListCommonPrefixes(bucket, joinPathPrefix(bucketPath, "/"))
```

Comment:
If `bucketPath` is not blank, this ends up listing e.g. `"foo/bar"`. Note there is no trailing `/`. Is that ok? I'm thinking we need the trailer? cc @skriss

Reply:
fixed